### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Fombi-Favour**
In this activity, we:
- setup the environment
- created class files for `Person`, `Student` and `Teacher`
- added constructors and methods for each file
- updated the readme file

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  # constructor
+  def initialize(id, age, parent_permision, name = 'unknown')
+    @id = id
```
Comment (omar-muhamad) [inline person.rb]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=parent_permission%20as%20parameter.-,name%20and%20parent_permission%20are%20optional%20and%20have%20default%20values%20of%20%22Unknown%22%20and%20true.,-Getters%20for%20%40id), The `parent_permission` parameter has a default value `true`. Please add it.

- [ ] [Optional] [Highly recommended] It's a good practice and professional to generate the `id` automatically like the suggestion below, not as a parameter as you did here.
```ruby
def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(id, age, parent_permision, classroom, name = 'unknown')
+    super(id, name, age, parent_permision)
```
Comment (omar-muhamad) [inline student.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=parent_permission%20as%20parameter.-,name%20and%20parent_permission%20are%20optional%20and%20have%20default%20values%20of%20%22Unknown%22%20and%20true.,-Getters%20for%20%40id), The `parent_permission` parameter has a default value `true`. Please add it.

- [ ] [Optional] [Highly recommended] If you implement the `id` suggestion above, please remove the `id` from lines 4 and 5.

[Hint] It should be like that after fixing:
```ruby
def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(id, age, parent_permision, specialization, name = 'unknown')
+    super(id, name, age, parent_permision)
```
Comment (omar-muhamad) [inline teacher.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=parent_permission%20as%20parameter.-,name%20and%20parent_permission%20are%20optional%20and%20have%20default%20values%20of%20%22Unknown%22%20and%20true.,-Getters%20for%20%40id), The `parent_permission` parameter has a default value `true`. Please add it.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Fombi-Favour,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb (hunk 1 above)
Comment (Fombi-Favour) [reply person.rb]: I followed the requirement by adding the default value of the parent_permission for all the files. But it gives me a linter error. That's why I removed it and left it without a default value

Comment (omar-muhamad) [reply person.rb]: Hi @Fombi-Favour,
try `parent_permission: true` instead of `parent_permission = true`

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  # constructor
+  def initialize(id, age, name = 'unknown', parent_permision: true)
+    @id = id
```
Comment (omar-muhamad) [inline person.rb:7]: - [ ] [Optional] [Highly recommended] It's a good practice and professional to generate the `id` automatically like the suggestion below, not as a parameter as you did here.
```ruby
def initialize(age, name = "Unknown", parent_permission: true)
    @id = Random.rand(1..1000)
    @age = age
```

File: student.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(id, age, classroom, name = 'unknown', parent_permision: true)
+    super(id, name, age, parent_permision)
```
Comment (omar-muhamad) [inline student.rb:5]: - [ ] [Optional] [Highly recommended] If you implement the `id` suggestion above, please remove the `id` from lines 4 and 5.

File: teacher.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(id, age, specialization, name = 'unknown', parent_permision: true)
+    super(id, name, age, parent_permision)
```
Comment (omar-muhamad) [inline teacher.rb:5]: - [ ] [Optional] [Highly recommended] Same here, If you implement the `id` suggestion above, please remove the `id` from lines 4 and 5.

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @Fombi-Favour,
#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 
### ⚠️Please consider the Optional suggestions below⚠️

robert-redford-nod

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- Please check the inline comments below ⬇.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by danmainah** [comment]
Hi @Fombi-Favour , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the project requirements.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**